### ISSUE TITLE
CodeCommit deprecated

### ISSUE THREAD

**Issue by yaolu-dtp** [awslabs/landing-zone-accelerator-on-aws#520]
**Is your feature request related to a problem? Please describe.**
CodeCommit is deprecated. https://aws.amazon.com/blogs/devops/how-to-migrate-your-aws-codecommit-repository-to-another-git-provider/

**Describe the feature you'd like**
Can we have instructions for using an alternative git repo?

**Additional context**
Nil

**Comment by ssennettau** [comment]
Confirmed by Jeff Barr on [Twitter](https://x.com/jeffbarr/status/1818461689920344321) specifically about the deprecation of CodeCommit.

This isn't just a feature request now - this change actively prevents the deployment of AWS Landing Zone Accelerator into new AWS Accounts.

**Comment by snemir2** [comment]
lol.   a while back i opened up a feature request to pull directly from github.    
It probably would be best to integrate LZA directly with Git Hub.

**Comment by n2aws** [comment]
> lol. a while back i opened up a feature request to pull directly from github. It probably would be best to integrate LZA directly with Git Hub.

I second this.   since the main LZA code base is hosted on github already,  it'd be ideal to at least support github for the configuration files too.  Ideally "any git provider" would be the preference, then users can do self hosted, github, gitlab, bitbucket, etc etc.

**Comment by vanja-zecevic** [comment]
I agree, would be much better to have native support for github, bitbucket, etc. for config files. Otherwise we will need to implement our own pipeline to pull from git to S3.

**Comment by kisst** [comment]
I created an LZA customisation which deploys a CFN template , contains an IAM role and using OICD it grants permission to the repository for zip upload and created a GitHub Actions which does the zip and upload to S3 .
As a result my pipeline is up to date with config hosted on GH side, however this is pushed config and not pulled like the src code for LZA which also needs a GH PAT.
All code is generalised so happy to share it with the public if you think it would be useful .

**Comment by samispurs** [comment]
When do we anticipate this will get resolved?  The code uses Github to pull, surely can't be that difficult to get this updated to allow Github to be used to store the yaml files?

**Comment by SennaSemakula** [comment]
Agreed with supporting Github. When we initially adopted LZA a year ago, we built inhouse tooling to have GitHub as the main  landing zone interface where it would sync changes to AWS Codecommit in the background. 

Having Github natively supported by LZA will allow us to remove additionally complexity.

**Comment by richardkeit** [comment]
@samispurs , @SennaSemakula - please review what others have done in regards to this here - https://github.com/awslabs/landing-zone-accelerator-on-aws/issues/612

**Comment by Athanasii** [comment]
This issue has been addressed in **Landing Zone Accelerator v1.9.0**.

**Resolution:**
- **Changelog Entry**: `feat(s3): added use of S3 as a configuration repository location`
- **Additional**: `feat(pipeline): add codeconnection as configuration source` (v1.10.0)

**What's Available:**
LZA now supports multiple alternatives to CodeCommit:
1. **Amazon S3** - Store configuration files in an S3 bucket (v1.9.0+)
2. **AWS CodeConnections** - Connect to GitHub, GitLab, or Bitbucket (v1.10.0+)

**Migration Path:**
Please refer to the LZA documentation for instructions on migrating from CodeCommit to S3 or CodeConnections.

**Action Required:**
Upgrade to LZA v1.9.0 or later and migrate your repository as CodeCommit has been deprecated by AWS.

**Reference:**
- v1.9.0 Release: https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.9.0
- v1.10.0 Release: https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.10.0

Closing this issue as alternatives to CodeCommit are now available. Thank you!
